src/platform/run_on_demand.rs

```diff
@@ -76,6 +76,14 @@ impl<T> EventLoopExtRunOnDemand for EventLoop<T> {
     where
         F: FnMut(Event<Self::UserEvent>, &EventLoopWindowTarget<Self::UserEvent>),
     {
+        self.event_loop.window_target().clear_exit();
```

Comment:
Nit: I think I'd prefer to be able to refactor things internally, maybe each platform should call this internally instead? Though I recognize that for some platforms, that'd mean setting the flag unnecessarily in `run` as well, so IDK.

Reply:
it's the same on every platform though and they all have the same ways of doing so.

Reply:
Yeah, but I think I might change that for macOS in the future ;)

Anyway, it's private API, so doesn't really matter.

Reply:
Well, the point is to make backend maintainers aware that they need to do such a thing.